lib/chat/actions.tsx

```diff
@@ -0,0 +1,509 @@
+import 'server-only'
```

Comment:
Could I ask you to explain why both 'import server-only' and 'use server' are used? It seems that both have similar effects, yet 'use server' offers the flexibility to be applied per function rather than per module.

Reply:
https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment